File: ulohy/22-denni-rozvrh-ii/reseni/main.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        scanner.nextLine();

        String[] activityNames = new String[n];
        int[] activityStartTimes = new int[n];
        int[] activityEndTimes = new int[n];

        for (int i = 0; i < n; i++) {
            scanner.nextLine();

            activityNames[i] = scanner.nextLine();

            activityStartTimes[i] = readTime(scanner);
            activityEndTimes[i] = readTime(scanner);

            scanner.nextLine();
        }

        int currentTime = readTime(scanner);

        boolean activityFound = false;
        for (int i = 0; i < n; i++) {
            int startTime = activityStartTimes[i];
            int endTime = activityEndTimes[i];

            if ((currentTime >= startTime && currentTime <= endTime)
                    || (startTime > endTime && (currentTime >= startTime || currentTime <= endTime))) {
                activityFound = true;
                System.out.println(activityNames[i]);
                break;
            }
        }

        if (!activityFound) {
            System.out.println("none");
        }
    }

    static int readTime(Scanner scanner) {
        int hours = scanner.nextInt();
        int minutes = scanner.nextInt();
        return hours * 60 + minutes;
    }
}

File: ulohy/22-denni-rozvrh-ii/reseni/main.cpp
#include <iostream>
#include <string>
#include <vector>

int readTime() {
    int hours, minutes;
    std::cin >> hours >> minutes;
    return hours * 60 + minutes;
}

int main() {
    int n;
    std::cin >> n;
    std::cin.ignore();

    std::vector<std::string> activityNames(n);
    std::vector<int> activityStartTimes(n);
    std::vector<int> activityEndTimes(n);

    for (int i = 0; i < n; i++) {
        std::cin.ignore();

        std::getline(std::cin, activityNames[i]);

        activityStartTimes[i] = readTime();
        activityEndTimes[i] = readTime();

        std::cin.ignore();
    }

    int currentTime = readTime();

    bool activityFound = false;
    for (int i = 0; i < n; i++) {
        int startTime = activityStartTimes[i];
        int endTime = activityEndTimes[i];

        if ((currentTime >= startTime && currentTime <= endTime)
            || (startTime > endTime && (currentTime >= startTime || currentTime <= endTime))) {
            activityFound = true;
            std::cout << activityNames[i] << std::endl;
            break;
        }
    }

    if (!activityFound) {
        std::cout << "none" << std::endl;
    }

    return 0;
}
